api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// +kubebuilder:validation:XValidation:rule="(has(self.allow) || has(self.deny))",message="one of allow or deny must be specified"
+//
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	// +kubebuilder:validation:MinItems=1
+	Rules []Rule `json:"rules,omitempty"`
+}
+
+// Rule defines the single authorization rule.
+type Rule struct {
+	// ClientSelector contains the client selector configuration.
+	ClientSelector ClientSelector `json:"clientSelector,omitempty"`
+
+	// Permission contains the permissions for the selector.
+	Permission Permission `json:"permissions,omitempty"`
+
+	// Action defines the action to be taken if the rule matches.
+	Action RuleActionType `json:"action,omitempty"`
+}
+
+// ClientSelector contains the client selector configuration.
+type ClientSelector struct {
+	// ClientCIDRs is a list of CIDRs.
+	// Valid examples are "192.168.1.0/24" or "2001:db8::/64"
+	//
+	// +optional
+	ClientCIDRs []string `json:"clientCIDR,omitempty"`
+}
+
+// Permission contains the permissions for the selector.
+type Permission struct {
+	// Any indicates that the rule matches any permission.
+	//
+	// +optional
+	Any bool `json:"any,omitempty"`
+}
+
+// RuleActionType specifies the types of authorization rule action.
+// +kubebuilder:validation:Enum=AllowRuleType;DenyRuleType;LogRuleType
+type RuleActionType string
+
+const (
+	// AllowRuleType is the action to allow the request.
+	AllowRuleType RuleActionType = "ALLOW"
```

Comment:
imo lets use `Allow`, `Deny`, `Log` to use the same casing as other fields in the API such as the RateLimit Unit field (`Hour, `Second` etc)